app/controllers/homepage_controller.rb

```diff
@@ -0,0 +1,19 @@
+class HomepageController < ApplicationController
+
+  def index
+    @products = Product.all
+
+    @products = @products.where(category: :entree) if params[:category] == 'entree'
+    @products = @products.where(category: :second_course) if params[:category] == 'second-course'
+    @products = @products.where(category: :dessert) if params[:category] == 'dessert'
+
+    @products = @products.order(:name) if params[:order] == 'az'
+    @products = @products.order(name: :desc) if params[:order] == 'za'
+    @products = @products.order(:price) if params[:order] == 'ascending'
+    @products = @products.order(price: :desc) if params[:order] == 'descending'
+
+    @products = @products.where(vegetarian: true) if params[:vegetarian] == 'vegetarian'
```

Comment:
Because the params you send are the same with the enum values or booleans, you could also use directly:

`@products.where(category: params[:category])` and `@products.where(vegetarian:  params[:vegetarian] == 'vegetarian')`